From 7a69570b2d5d6011439cc8afb1758f0c3e1b18bc Mon Sep 17 00:00:00 2001
Message-Id: <7a69570b2d5d6011439cc8afb1758f0c3e1b18bc.1241177871.git.joerg@alea.gnuu.de>
In-Reply-To: <e879f68a77182a30e0ee2c1ffaea9ae9cd12f326.1241177871.git.joerg@alea.gnuu.de>
References: <e879f68a77182a30e0ee2c1ffaea9ae9cd12f326.1241177871.git.joerg@alea.gnuu.de>
From: =?utf-8?q?J=C3=B6rg=20Sommer?= <joerg@alea.gnuu.de>
Date: Fri, 1 May 2009 13:25:19 +0200
Subject: [PATCH 2/4] main: Ignore the last line of a diversion triplet
MIME-Version: 1.0
Content-Type: text/plain; charset=utf-8
Content-Transfer-Encoding: 8bit

Dpkg prints in the last line for a diversion a summary with both packages
they take part on the diversion. Therefore, the split at the colon
doesn't work, because two packages are names before the colon. But the
package for the diversion was put into the list in the line “diversion …
to”. So we can skip this line.

% dpkg -S /usr/bin/mutt
diversion by mutt-patched from: /usr/bin/mutt
diversion by mutt-patched to: /usr/bin/mutt.org
mutt-patched, mutt: /usr/bin/mutt

Otherwise this error message is printed:

Package `mutt-patched, mutt' is not installed.
Use dpkg --info (= dpkg-deb --info) to examine archive files,
and dpkg --contents (= dpkg-deb --contents) to list their contents.
---
 checkrestart |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/checkrestart b/checkrestart
index bbea0c8..baef62f 100755
--- a/checkrestart
+++ b/checkrestart
@@ -149,6 +149,11 @@ def main():
             packagename, program = m.group(1), diverted
         else:
             packagename, program = line[:-1].split(': ')
+            if program == diverted:
+                # dpkg prints a summary line after the diversion, name both
+                # packages of the diversion, so ignore this line
+                # mutt-patched, mutt: /usr/bin/mutt
+                continue
 
         packages.setdefault(packagename,Package(packagename))
         packages[packagename].processes.extend(programs[program])
-- 
1.6.3.rc1

